config/initializers/0_clowder-config.rb

```diff
@@ -1,19 +1,33 @@
 
+def build_endpoint_url ep
+  config = ClowderCommonRuby::Config.load
```

Comment:
You are loading the already loaded clowder config on each call of this method. Please pass the config as a parameter from the already loaded instance.